scripts/competitiond.py

```diff
@@ -738,10 +743,19 @@ def run_once(self):
 
         if not self.leaderboard_only:
             for submit_bundle in submissions:
-                logger.info(
-                    "Mimicking submission for " "{owner[user_name]}".format(**submit_bundle)
-                )
-                predict_bundle = self.run_prediction(submit_bundle)
+                if 'user_name' in submit_bundle['owner']:
+                    logger.info(
+                        "Mimicking submission for " "{owner[user_name]}".format(**submit_bundle)
+                    )
+                else:
+                    logger.info(
+                        "Mimicking submission for " "{metadata[description]} (no username)".format(**submit_bundle)
+                    )
+                try:
+                    predict_bundle = self.run_prediction(submit_bundle)
+                except Exception as e:
+                    traceback.print_exc()
```

Comment:
Should we also log something like this?

```suggestion
                    traceback.print_exc()
                    logger.info(
                        "Aborting submission for " "{owner[user_name]}".format(**submit_bundle)
                    )
```

Reply:
It's possible for the user_name field to not exist (that's why I added the if statement above), so we'd need a similar if statement here.